app/models/branch.rb

```diff
@@ -0,0 +1,12 @@
+# Model class Branch
+class Branch < ApplicationRecord
+  validates :name, :opening_time, :closing_time, presence: true
+  validates :closing_time, comparison: { greater_than: :opening_time }, if: :closing_time
+  validate :only_one_default, if: :default
```

Comment:
Add a check before destroy that there should be at least one default branch.
so before deleting a default branch, we have to make another default.

Reply:
and while updating default bit `false` as well